target/cheri-common/cheri-translate-utils.h

```diff
@@ -88,30 +90,43 @@ static inline bool have_cheri_tb_flags(DisasContext *ctx, uint32_t flags)
     return (ctx->base.cheri_flags & flags) == flags;
 }
 
-static inline void _generate_ddc_checked_ptr(
+// Checks an address against PCC or DDC.
+// Permissions are checked at translate time
+// Bounds checks can be skipped if CHERI flags indicate bounds would make checks
+// trivial.
+static inline void _generate_special_checked_ptr(
     DisasContext *ctx, CheriTbFlags tb_perm_flags, CheriPermissions req_perms,
-    TCGv_cap_checked_ptr checked_addr, TCGv ddc_offset, target_ulong num_bytes)
+    TCGv_cap_checked_ptr checked_addr, TCGv ddc_offset, target_ulong num_bytes,
```

Comment:
Blegh `ddc_offset` should be renamed, both because it's now also for PCC and because it's not always the offset (in the PCC case and in the TB_FLAG_CHERI_DDC_NO_INTERPOSE case).